astropy/table/table.py

```diff
@@ -1618,12 +1662,58 @@ def add_columns(self, cols, indexes=None, copy=True, rename_duplicate=False):
               2 0.2 1.2  y
               3 0.3 1.3  z
 
+        Add unnamed columns or mixin objects at the end of the table using default names::
+            >>> t = Table([[1, 2, 3], [0.1, 0.2, 0.3]], names=('a', 'b'))
```

Comment:
Actually, just leave it as is, that's fine.  What I would suggest, though, is to combine the examples here, so that the docstring does not become so very long. I think in the end this is a fairly simple function, and there is a benefit to just having the documentation fit in a single screen. Speficically, you could just add the columns twice, once without and once with the names given, and then show a single print statement. For brevity, I'd also use just two elements.

p.s. And of course do the same for `add_column` above.